dev/ToastNotifications/ToastNotificationUtility.cpp

```diff
@@ -212,3 +212,80 @@ std::wstring RegisterComActivatorGuidAndAssets(winrt::Microsoft::Windows::ToastN
     THROW_IF_WIN32_ERROR(status);
     return registeredGuid;
 }
+
+wil::unique_cotaskmem_string ConvertUtf8StringToWideString(unsigned long length, const byte* utf8String)
+{
+    int size{ MultiByteToWideChar(
+        CP_UTF8,
+        0,
+        reinterpret_cast<PCSTR>(utf8String),
+        length,
+        nullptr,
+        0) };
+    THROW_LAST_ERROR_IF(size == 0);
+
+    wil::unique_cotaskmem_string wideString{ wil::make_unique_string<wil::unique_cotaskmem_string>(nullptr, size) };
+
+    size = MultiByteToWideChar(
+        CP_UTF8,
+        0,
+        reinterpret_cast<PCSTR>(utf8String),
+        length,
+        wideString.get(),
+        size);
+    THROW_LAST_ERROR_IF(size == 0);
+
+    return wideString;
+}
+
+winrt::Microsoft::Windows::ToastNotifications::ToastNotification ToastNotificationFromToastProperties(ABI::Microsoft::Internal::ToastNotifications::INotificationProperties* properties)
+{
+    unsigned int payloadSize{};
+    wil::unique_cotaskmem_array_ptr<byte> payload{};
+    THROW_IF_FAILED(properties->get_Payload(&payloadSize, &payload));
+
+    auto wide{ ConvertUtf8StringToWideString(payloadSize, payload.get()) };
+    winrt::hstring xmlPayload{ wide.get() };
+
+    winrt::Windows::Data::Xml::Dom::XmlDocument xmlDocument{};
+    xmlDocument.LoadXml(xmlPayload);
+
+    winrt::Microsoft::Windows::ToastNotifications::ToastNotification notification(xmlDocument);
+
+    wil::unique_hstring tag{};
+    properties->get_Tag(&tag);
+    notification.Tag(wil::str_raw_ptr(tag));
+
+    wil::unique_hstring group{};
+    properties->get_Group(&group);
+    notification.Group(wil::str_raw_ptr(group));
+
+    unsigned int notificationId{};
+    properties->get_NotificationId(&notificationId);
+    notification.ToastId(notificationId);
+
+#if 0
```

Comment:
Whats this test code doing here?

Reply:
That  was a place holder waiting for Sharath's the progress data stuff to be completed. Code has been adapted to account for progress data now.